tests/test_crawlers.py

```diff
@@ -6,12 +6,37 @@
 import unittest
 import unittest.mock as mock
 from datetime import datetime, timezone
+import geospaas_harvesting.harvest as harvest
 
 import requests
 
 import geospaas_harvesting.crawlers as crawlers
 
 
+class WebDirectoryCrawlerExceptionTestCase(unittest.TestCase):
+    """Tests for the webDirectory crawler Exceptions"""
+
+    def test_lack_of_definition_of_get_download_url(self):
+        """shall return "NotImplementedError" exception in the case of lack of definition of get_download_url method """
+        my_harvester = crawlers.WebDirectoryCrawler
+        with self.assertRaises(NotImplementedError):
+            my_harvester.get_download_url("test_text", "test_text2")
+
+    def test_lack_of_definition_of_set_initial_state(self):
+        """shall return "NotImplementedError" exception in the case of lack of definition of set_initial_state method """
+        with self.assertRaises(NotImplementedError):
+            crawlers.Crawler.set_initial_state(self)
+
+    @mock.patch("geospaas_harvesting.crawlers.OpenDAPCrawler._http_get")
+    @mock.patch("geospaas_harvesting.crawlers.OpenDAPCrawler._get_links")
+    def test_lack_of_correct_url_for_download_page_for_OSISAF(self, mock_get_link, mock_http_get):
+        """Test the functionality of "get_download_url" method for OpenDAP crawler with incorrect
+        url. URL must be ended with '.html' """
+        mock_get_link.return_value = ['/thredds/dodsC/osisaf/met.no/ice_conc201911301200.nc']
+        with self.assertLogs(crawlers.OpenDAPCrawler.LOGGER):
+            crawlers.ThreddsCrawler.get_download_url(crawlers.OpenDAPCrawler, "dummy")
```

Comment:
Same comment as https://github.com/nansencenter/django-geo-spaas-harvesting/pull/20/files#r472876823

Reply:
Done in the next commit